plutus-emulator/src/Wallet/Emulator/Types.hs

```diff
@@ -267,6 +266,11 @@ handleNotifications = mapM_ (updateState >=> runTriggers)  where
     -- Remove spent outputs and add unspent ones, for the addresses that we care about
     update t = over addressMap (AM.updateAddresses t)
 
+-- Make a transaction output from a positive value.
+mkChangeOutput :: PubKey -> Value -> Maybe TxOut
+mkChangeOutput pubK v =
+    if Value.isZero v then Nothing else Just (pubKeyTxOut v pubK)
```

Comment:
Should check for < 0? The calling code will never do this, but I think this should still return `Nothing` in that case.

Reply:
To be more specific, should we check that all token names are positive?

Reply:
Then it would probably make sense to add a predicate in Value, something like `thereIsNoNegativeTokenValues :: Value -> Boolean` in Value, maybe add a new issue to check that we don't have any negative values embedded in transactions? Like a validity check?

Reply:
`Value` deliberately allows non-positive values. This makes it simple to implement subtraction, etc., and enables us to have negative values in `forge` fields, which allows us to destroy currency, which want to do.

We could try and be clever in the type system about requiring other `Value`s to be positive, but this is enforced by the ledger rules (we should check that our implementation actually does this), so transactions will just fail to validate if you put negative values in, which seem fine for now.

Reply:
There is a check for negative values here: https://github.com/input-output-hk/plutus/blob/master/plutus-wallet-api/ledger/Ledger/Index.hs#L267